op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,407 @@
+	broadcast chan []byte
+
+	// Signal to stop the hub
+	done chan struct{}
+}
+
+// newHub creates a new hub
+func newHub() *Hub {
+	return &Hub{
+		broadcast:  make(chan []byte, BroadcastBufferSize),
+		register:   make(chan *Client),
+		unregister: make(chan *Client, UnregisterBufferSize),
+		clients:    make(map[*Client]bool),
+		done:       make(chan struct{}),
+	}
+}
+
+// run starts the hub's main loop
+func (h *Hub) run() {
+	for {
+		select {
+		case <-h.done:
+			// Close all client connections
+			h.clientsMu.Lock()
+			for client := range h.clients {
+				client.Close()
+				delete(h.clients, client)
+			}
+			h.clientsMu.Unlock()
+			return
+		case client := <-h.register:
+			h.clientsMu.Lock()
+			h.clients[client] = true
+			clientCount := len(h.clients)
+			h.clientsMu.Unlock()
+			log.Info("Client registered with hub", "remote", client.conn.RemoteAddr(), "totalClients", clientCount)
+		case client := <-h.unregister:
+			h.clientsMu.Lock()
+			if _, ok := h.clients[client]; ok {
+				delete(h.clients, client)
+				client.Close()
+			}
+			h.clientsMu.Unlock()
+		case message := <-h.broadcast:
+			h.clientsMu.RLock()
+			for client := range h.clients {
+				select {
+				case client.send <- message:
+					// Message sent successfully
+					client.failureCount = 0 // Reset failure count on success
+				default:
+					// Client send buffer is full, increment failure count
+					client.failureCount++
+					if client.failureCount >= MaxClientFailures {
+						// Try to unregister, but don't block if channel is full
+						select {
+						case h.unregister <- client:
+							// Successfully queued for unregistration
+						default:
+							// Unregister channel is full, we'll try again next loop
+							log.Warn("Unregister channel full, client will be retried next loop",
+								"remote", client.conn.RemoteAddr())
+						}
+					}
+				}
+			}
+			h.clientsMu.RUnlock()
+		}
+	}
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	hub             *Hub
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) (FlashblockHandler, error) {
+	// Create the hub
+	hub := newHub()
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		hub:        hub,
+	}
+
+	// Establish connection to rollup boost if URL is configured
+	if cfg.RollupBoostWsURL == "" {
+		log.Info("rollup boost WebSocket disabled, no URL configured")
+		return handler, nil
+	}
+
+	log.Info("connecting to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+	conn, _, err := websocket.DefaultDialer.Dial(cfg.RollupBoostWsURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+	}
```

Comment:
better to retry here (maybe fixed interval 10s, retry for 6 times), just in case rollup-boost hasn't been started properly